File: 169. Majority Element/majorityElement.cpp
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {

        unordered_map<int, int> m;
        for (auto c : nums)
        {
            m[c]++;
        }
        priority_queue<pair<int, int>> maxh;
        for (auto it : m)
        {
            maxh.push({it.second, it.first});
        }
        int pivot = nums.size() / 2;
        while (maxh.size() > 0)
        {
            if (maxh.top().first > pivot)
                return maxh.top().second;
        }
        return -1;
    }
};

File: 169. Majority Element/majorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
        
        int n = nums.length;
        int pivot = n/2;
        Map<Integer,Integer> m = new HashMap<>();
        for(int i=0;i<n;i++){
            int val = m.getOrDefault(nums[i],0);
            m.put(nums[i],val+1);
        }
        for(Map.Entry <Integer,Integer> it : m.entrySet()){
            if(it.getValue() > pivot) return it.getKey();
             
        }
        return -1;
    }
}